lib/taiko.js

```diff
@@ -1660,8 +1660,41 @@ async function highlight(selector, ...args) {
  */
 module.exports.mouseAction = mouseAction;
 
-async function mouseAction(action, coordinates, options = {}) {
+async function mouseAction(
+  selector,
+  action,
+  coordinates,
+  options = {},
+) {
   validate();
+  var actions = ['press', 'move', 'release'];
+  if (
+    (isSelector(selector) ||
+      isString(selector) ||
+      Number.isInteger(selector)) &&
+    !actions.includes(selector)
+  ) {
+    const elem = await element(selector);
+    if (elem == null) {
+      throw Error(
+        'Please provide a valid selector, unable to find element',
+      );
+    }
+    await scrollTo(elem);
+    let {
+      top,
+      bottom,
+      left,
+      right,
+    } = await domHandler.getBoundingClientRect(elem);
+    coordinates.x = coordinates.x + parseInt(left);
+    coordinates.y = coordinates.y + parseInt(top);
+  } else if (actions.includes(selector)) {
+    coordinates = action;
+    action = selector;
+  } else {
+    throw Error('Please provide a valid parameter');
```

Comment:
the error message can perhaps be a little more specific. for instance in this case there are the following checks:

```
isSelector(selector) ||
      isString(selector) ||
      Number.isInteger(selector)) &&
    !actions.includes(selector)
```

if this was refactored to have an inverted `if` then the error message can pass specific details. ex.

```
if (!isSelector(selector)) throw Error('not a valid selector: ' + selector);
if (!isString(selector)) throw Error('expected selector to be of type 'String': ' + selector)
...
etc
...
```

Reply:
sure @sriv i will do that and commit it